New arguments and speed improvements - Version 1.1.4

Skinflint opened the pull request:
Added arguments to get the version of the script, update the installomator and patchomator scripts if they need updated, set a script proxy if necessary, and to use SPCTL instead of codesign if needing more verification on apps. 

Changed the version check of the app labels to use 'is-at-least'.

Rearranged installomator check and added a fatal check if installomator is not installed and the script is being run in install mode.

Created some arrays for the labels to not require a second lookup if needed.

Added a script check to the verifyApp function. If the script is installed from GitHub it downloads the latest pkg or the version installed pkg and does an md5 file check and a spctl or pkgutil check for expectedTeamID.

Added a summary after the appVerify that will let the user know how many of the apps that were found need updates. Only available if not in quiet mode.

Updated the DialogPath and added a line to remove the file. Also added more quiet mode checks for dialog.

Added a lock file to prevent the script from running if it is already running. It also checks that file for the PID in case the file is there but the script isn't already running.

Added a trap for the code if the script exits abnormally. It will kill the caffeinate and dialog processes.

The script now runs discovery before installs. It will check the appversion of the found apps and only send ones that need updates to installomator. If discovery wants to be skipped, then you will need to use the --ignored "all" command.

Updates to the readme.md

Miscellaneous cleanup mostly involving tabs and spaces.

Skinflint committed:
New arguments and speed improvements

Added arguments to get the version of the script, update the installomator and patchomator scripts if they need updated, and to set a script proxy if necessary.

Changed the version check of the app labels to use 'is-at-least'.

Rearranged installomator check and added a fatal check if installomator is not installed and the script is being run in install mode.

Created some arrays for the labels to not require a second lookup if needed.

Added a script check to the verifyApp function. If the script is installed from GitHub it downloads the latest pkg or the version installed pkg and does an md5 file check and a spctl or pkgutil check for expectedTeamID.

Added a summary after the appVerify that will let the user know how many of the apps that were found need updates. Only available if not in quiet mode.

Miscellaneous cleanup mostly involving tabs and spaces.

Skinflint committed:
Move install check before file check

Need to set all options from the install check before reading from the config file if it exists

Skinflint committed:
Fix required app skips other apps

Add a new array to fix the required apps not being the label assigned to the application.

Skinflint committed:
Add more quiet checks and squash a bug

Only use dialog if not in quiet mode, and fix quoting bug for the required labels.

Skinflint committed:
DialogPath Update

Updated the DialogPath and added a line to remove the file

Skinflint committed:
Grab installomator exit status

Get the installomator exit status into a variable.

Skinflint committed:
Moved quit dialog to finishandexit

Moved the echo of quit for the dialog log file to the finishAndExit function to remove extra lines.

Skinflint committed:
Dialog Log File Check

Added a check that the dialog log file is writable and created after it is removed. Moved the finishandexit out of the display config function.

Skinflint committed:
Allow discovery and install mode

Allow Discovery and install mode at the same time

Skinflint committed:
Update README.md

Update readme for new switches and updated command output.

Skinflint committed:
Update README.md

Skinflint committed:
Fix replace label issues

Fixed some issues around the replacement of apps if not using --yes. It was adding extra unique apps and not removing from the update list.

Skinflint committed:
Fix update tally

Remove the item from the appNeedsUpdates tally if it is being replaced with a different label

Skinflint committed:
Add Cleanup

Added some cleanup for if the scripts ends improperly. Also added caffeinate to the discovery.

Skinflint committed:
Fix queuedLabelsArray

Use the installedLabelsList instead of labelsList for the queuedLabelsArray, and remove teh @s/ / so that an empty array doesn't add a blank and give a count of 1.

Skinflint committed:
Continue on ALL

Continue on ALL instead of break so that if there is a list of ignores on the command line, it can remove them from the install list.

Skinflint committed:
Update README.md

Skinflint committed:
Add a lockfile

Add a lockfile that includes the running PID to prevent the script from being ran if it is already running.

Skinflint committed:
patchomatorPATH

Get patchomatorPATH with fallback to default location.

Mac-Nerd commented on patchomator.sh:
```diff
@@ -157,28 +163,37 @@ YELLOW=$(tput setaf 3 2>/dev/null)
+	echo "\tScans the system for installed apps and matches them to Installomator labels.\n"
+	echo "\t${BOLD}--version \t${RESET} Show version and exit."
+	echo "\t${BOLD}--fullversion \t${RESET} Show full version and exit."
+	echo "\t${BOLD}--proxy \"proxyIP:Port\" \t${RESET} Show version and exit."
```
proxy help text

Mac-Nerd commented on patchomator.sh:
```diff
@@ -157,28 +163,37 @@ YELLOW=$(tput setaf 3 2>/dev/null)
+	echo "\t${BOLD}--fullversion \t${RESET} Show full version and exit."
+	echo "\t${BOLD}--proxy \"proxyIP:Port\" \t${RESET} Show version and exit."
+	echo "\t${BOLD}--required \"space-separated list of labels to require\""
+	echo "\t${BOLD}--ignored \"space-separated list of labels to ignore\"${RESET}\n\t\t If list contains ${YELLOW}'ALL'${RESET} then discovery will be skipped\n\t\t If list contains ${YELLOW}'RECOMMENDEDIGNORES'${RESET} then the recommended list of ignores will be appended.\n"
```
RECOMMENDEDIGNORES to RECOMMENDED

Mac-Nerd commented on patchomator.sh:
```diff
@@ -777,54 +888,74 @@ queueLabel() {
+	echo "$VERSIONDATE - $VERSION"
+	echo "$VERSIONNAME"
+	exit 0
+fi
```
added installomator version, formatting

Mac-Nerd commented on patchomator.sh:
```diff
@@ -1114,42 +1228,43 @@ if [[ -n "$ignoredLabels" ]]
+			skipDiscovery=true
+			break
+		fi
+		if [[ "$lowerLabel" == "recommendedignores" ]]; then
```
changed option RECOMMENDEDIGNORES to RECOMMENDED

Skinflint committed:
Recommended fixes

Changed RECOMMENDEDIGNORES to RECOMMENDED.

Changed the output of fullversion to closer match Installomator.

Fixed the proxy help text.

Skinflint committed:
Update README.md

Update for recommended list

Skinflint committed:
Fix RecommenedLabelsArray Quoting

Fixed and issue where quotes were not the same between reading from the config file and reading from the command line.

Skinflint committed:
Dialog Updates, Timeouts, and MDM

Added timeout for label replacement prompts and included a prompt via Swift Dialog. Default 120, can be set via --timeout (10-9000)

Added the --icon argument for setting the Swift Dialog icon.

Fixed the moveable argument for Swift Dialog to allow the command to be moved around.

Revamped the config file logic.

Added MDM option to set the Logos. More settings can be added in the future to that logic in the future.

Revamped the CLIOptions to allow for them to overwrite default options, and made all default options get pushed to Installomator even if CLIOptions exists.

Skinflint committed:
Installomator Prompts

Update Timeout and IAMROOT logic

Add timeout and dialog prompt for Installomator install
